src/metabase/query_processor/util.clj

```diff
@@ -57,6 +57,33 @@
   (default-query->remark query))
 
 
+;;; ------------------------------------------------- User Impersonation --------------------------------------------------
+
+;; TODO (noahmoss) find better namespace for this?
```

Comment:
Still not sure where these multimethod definitions should ideally go

Reply:
Definitely should not go in here... driver methods go in metabase.driver.* namesPaces. Since these are basically SQL-specific but maybe don't conceptually have to be JDBC-specific, you should probably put them in metabase.driver.sql .

Reply:
Or maybe put it in metabase.driver.sql-jdbc because it's currently only supported for JDBC-based DBs. It would be nice if the API design wasn't tied to a SQL and there was a way to implement this for another DB like Mongo if it supported it... I think what you really want to do is have a set-role! method in metabase.driver and then in driver.sql-jdbc you can have the set-role-statement method and a default implementation for set-role! that uses it

Reply:
Makes sense